src/Stryker.Core/Stryker.Core/Mutants/CsharpNodeOrchestrators/CommentParser.cs

```diff
@@ -65,32 +66,55 @@ private static MutationContext ParseStrykerComment(MutationContext context, Matc
         return context.FilterMutators(disable, filteredMutators, match.Groups[OnceGroup].Value.ToLower() == "once", comment);
     }
 
-    public static MutationContext ParseNodeComments(SyntaxNode node, MutationContext context)
+    public static MutationContext ParseNodeLeadingComments(SyntaxNode node, MutationContext context)
     {
-        foreach (var commentTrivia in node.GetLeadingTrivia()
-                     .Where(t => t.IsKind(SyntaxKind.SingleLineCommentTrivia) ||
-                                 t.IsKind(SyntaxKind.MultiLineCommentTrivia)).Select(t => t.ToString()))
+        // parse comment that is on a dedicated line before the syntx node
+        if (node.GetLeadingTrivia()
+            .Where(t => t.IsKind(SyntaxKind.SingleLineCommentTrivia) ||
+                        t.IsKind(SyntaxKind.MultiLineCommentTrivia)).Select(t => t.ToString()).Any(commentTrivia => ProcessComment(node, ref context, commentTrivia)))
         {
-            // perform a quick pattern check to see if it is a 'Stryker comment'
-            if (!Pattern.Match(commentTrivia).Success)
-            {
-                continue;
-            }
-
-            var match = Parser.Match(commentTrivia);
-            if (match.Success)
-            {
-                // this is a Stryker comments, now we parse it
-                context = ParseStrykerComment(context, match, node);
-                break;
-            }
+            return context;
+        }
 
-            Logger.LogWarning(
-                "Invalid Stryker comments at {Position}, {FilePath}.",
-                node.GetLocation().GetMappedLineSpan().StartLinePosition,
-                node.SyntaxTree.FilePath);
+        // identify the previous syntax node if possible
+        var previousSyntaxNode = node.GetPreviousSyntaxNode();
+        if (previousSyntaxNode == null)
+        {
+            return context;
         }
+        // parse any trailing comment, assuming the user intend it to apply to this node.
+        previousSyntaxNode.GetTrailingTrivia()
+            .Where(t => t.IsKind(SyntaxKind.MultiLineCommentTrivia)).Select(t => t.ToString()).
```

Comment:
Should we include single-line comments here as well, or is there a reason we wouldn't want to support that scenario?

Reply:
thanks for the remark, it is just that I wrongly assumed that single line comment could not be trailing (which is a stupid idea in hindsight).